From 0c52baebe72a616a211b386c93afd0e67618764d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 11 Sep 2018 15:36:35 +0200
Subject: [PATCH] inputMethod: Fix setting surrounding text

The underlying ibus method expects an object of type IBusText rather
than a plain string.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/228
---
 js/misc/inputMethod.js | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/js/misc/inputMethod.js b/js/misc/inputMethod.js
index ec84f7277..7fb78178a 100644
--- a/js/misc/inputMethod.js
+++ b/js/misc/inputMethod.js
@@ -176,8 +176,11 @@ var InputMethod = new Lang.Class({
     },
 
     vfunc_set_surrounding(text, cursor, anchor) {
-        if (this._context && text)
-            this._context.set_surrounding_text(text, cursor, anchor);
+        if (!this._context || !text)
+            return;
+
+        let ibusText = IBus.Text.new_from_string(text);
+        this._context.set_surrounding_text(ibusText, cursor, anchor);
     },
 
     vfunc_update_content_hints(hints) {
-- 
2.18.0

